### PR TITLE
tests: options valid or invalid values are not comprehensive

### PR DESCRIPTION
**Pull request by Milly**

### CODE REVIEW

File: src/testdir/gen_opt_test.vim [hunk 1]
```diff
@@ -30,90 +30,153 @@ let fontname = has('win32') ? 'fixedsys' : 'fixed'
 
 " Two lists with values: values that work and values that fail.
 " When not listed, "othernum" or "otherstring" is used.
+" When both lists are empty, skip tests for the option.
+" For boolean options, if non-empty a fixed test will be run, otherwise skipped.
 let test_values = {
+      \ '### boolean options': '###############',
```
Comment (zeertzjq) [inline src/testdir/gen_opt_test.vim]:
```suggestion
      "\ boolean options
```

File: src/testdir/gen_opt_test.vim [hunk 2]
```diff
@@ -30,90 +30,153 @@ let fontname = has('win32') ? 'fixedsys' : 'fixed'
...
+      \ 'termguicolors': [
+      \		has('vtp') && !has('vcon') && !has('gui_running') ? [] : [1],
+      \		[]],
+      \
+      \ '### number options': '###############',
```
Comment (zeertzjq) [inline src/testdir/gen_opt_test.vim]:
```suggestion
      "\ number options
```

File: src/testdir/gen_opt_test.vim [hunk 3]
```diff
@@ -30,90 +30,153 @@ let fontname = has('win32') ? 'fixedsys' : 'fixed'
...
-      \		['', 'xxx', '<xxx>', '<Esc', '<t_xx', '<C-C>', '<NL>', '<CR>', K_KENTER]],
+      \		['', 'xxx', '<xxx>', '<t_xxx>', '<Esc', '<t_xx', '<C-C>',
+      \		'<NL>', '<CR>', K_KENTER]],
       \ 'winheight': [[1, 10, 999], [-1, 0]],
       \ 'winminheight': [[0, 1], [-1]],
       \ 'winminwidth': [[0, 1, 10], [-1]],
       \ 'winwidth': [[1, 10, 999], [-1, 0]],
       \
-      \ 'ambiwidth': [['', 'single'], ['xxx']],
+      \ '### string options': '###############',
```
Comment (zeertzjq) [inline src/testdir/gen_opt_test.vim]:
```suggestion
      "\ string options
```

File: src/testdir/gen_opt_test.vim [hunk 4]
```diff
@@ -122,78 +185,118 @@ let test_values = {
...
+      \		'bs'],
+      \		['xxx']],
+      \ 'wildmode': [['', 'full', 'longest', 'list', 'lastused', 'list:full',
+      \		'full,longest', 'full,full,full,full'],
+      \		['xxx', 'a4', 'full,full,full,full,full']],
       \ 'wildoptions': [['', 'tagfile', 'pum', 'fuzzy'], ['xxx']],
-      \ 'winaltkeys': [['menu', 'no'], ['', 'xxx']],
+      \ 'winaltkeys': [['no', 'yes', 'menu'], ['', 'xxx']],
       \
+      \ '### skipped options': '###############',
```
Comment (zeertzjq) [inline src/testdir/gen_opt_test.vim]:
```suggestion
      "\ skipped options
```

File: src/testdir/gen_opt_test.vim [hunk 5]
```diff
@@ -122,78 +185,118 @@ let test_values = {
...
       \ 'perldll': [[], []],
       \ 'pythondll': [[], []],
       \ 'pythonthreedll': [[], []],
       \ 'pyxversion': [[], []],
       \ 'rubydll': [[], []],
       \ 'tcldll': [[], []],
+      \ 'term': [[], []],
+      \ 'ttytype': [[], []],
       \
+      \ '### default behaviours': '###############',
```
Comment (zeertzjq) [inline src/testdir/gen_opt_test.vim]:
```suggestion
      "\ default behaviours
```

**Review by zeertzjq — COMMENTED** [review]
Any reason for adding a special comment syntax for `test_values` instead of using proper comments?

File: src/testdir/gen_opt_test.vim (hunk 1 above)
Comment (Milly) [reply src/testdir/gen_opt_test.vim]: Thanks!

**Comment by chrisbra** [comment]
Nice, thank you!

**Comment by zeertzjq** [comment]
It seems that `Test_writedelay()` is constantly failing in the `testgui` job in this PR:
```
Failures: 
	From test_options.vim:
	Found errors in Test_writedelay():
	command line..script /home/runner/work/vim/vim/src/testdir/runtest.vim[607]..function RunTheTest[57]..Test_writedelay line 13: Expected range 0.3 - 999.0, but got 0.002673
```

**Pull request closed without merging**